docs/reference/composition.md

```diff
@@ -510,6 +510,22 @@ You can use the following types of transform on a value being patched:
     au-east: Australia East
 ```
 
+`match`. A more complex version of `map` that can match different kinds of
+patterns.
```

Comment:
Some docs thoughts:

* Should we provide guidance on when to use map vs match? I imagine this could supersede/deprecate match if we wanted.
* I'd like us to document what happens when there are multiple matches.

Reply:
Good catch. I added to the description that `match` should be used if more advanced pattern matching techniques are required. I also stated that the result of the first matching pattern is picked as output of the transform.